nano/core_test/active_transactions.cpp

```diff
@@ -967,6 +967,89 @@ TEST (active_transactions, confirmation_consistency)
 }
 }
 
+/**
+ * This test case tests the creation of reverse links for incoming confirmed blocks.
+ */
+TEST (active_transactions, reverse_link)
+{
+	// Initialise node environment and enable reverse links
+	nano::system system;
+	nano::node_config node_config = nano::node_config (nano::get_available_port (), system.logging);
+	node_config.enable_reverse_links = true;
+	auto & node = *system.add_node (node_config);
+
+	// Generate a random key pair
+	nano::keypair key;
+	// Create two blocks
+	nano::state_block_builder builder;
+	auto send1 = builder.make_block ()
+				 .account (nano::dev::genesis_key.pub)
+				 .previous (nano::dev::genesis->hash ())
+				 .representative (nano::dev::genesis_key.pub)
+				 .balance (nano::dev::constants.genesis_amount - 1)
+				 .link (key.pub)
+				 .sign (nano::dev::genesis_key.prv, nano::dev::genesis_key.pub)
+				 .work (*system.work.generate (nano::dev::genesis->hash ()))
+				 .build_shared ();
+	auto open1 = builder.make_block ()
+				 .account (key.pub)
+				 .previous (0)
+				 .representative (key.pub)
+				 .balance (1)
+				 .link (send1->hash ())
+				 .sign (key.prv, key.pub)
+				 .work (*system.work.generate (key.pub))
+				 .build_shared ();
+	// Process and confirm both blocks
+	node.process_active (send1);
+	node.process_active (open1);
+	auto vote = std::make_shared<nano::vote> (nano::dev::genesis_key.pub, nano::dev::genesis_key.prv, nano::vote::timestamp_max, nano::vote::duration_max, std::vector<nano::block_hash>{ send1->hash (), open1->hash () });
+	node.vote_processor.vote (vote, std::make_shared<nano::transport::channel_loopback> (node));
+	ASSERT_TIMELY (5s, node.ledger.cache.cemented_count == 3);
+	// Check if reverse link was created
+	{
+		auto transaction = node.store.tx_begin_read ();
+		ASSERT_EQ (1, node.store.reverse_link.count (transaction));
```

Comment:
You cannot depend on count being accurate when using rocksdb.
I tend to do a manual count one by one by iterating through all the elements.
See function `manually_count_pruned_blocks`.

Reply:
> You cannot depend on count being accurate when using rocksdb. I tend to do a manual count one by one by iterating through all the elements. See function `manually_count_pruned_blocks`.

I guess that depends on the count implementation in `nano::rocksdb_store::count`? Some tables use `db->GetIntProperty (table_to_column_family (table_a), "rocksdb.estimate-num-keys", &sum);` as an approximation, because it is faster. But since the count of reverse links is only used in test cases with a rather small amount of elements, I decided to implement the exact counting:

```
else if (table_a == tables::reverse_links)
{
	for (auto i (reverse_link.begin (transaction_a)), n (reverse_link.end ()); i != n; ++i)
	{
		++sum;
	}
}
```

Reply:
I think it would be better to follow the existing pattern we already have.
The count of a store is unreliable. And unit tests use a different count function that counts one by one and is reliable.

Reply:
> I think it would be better to follow the existing pattern we already have for consistency. The count of a store is unreliable. And unit tests use a different count function that counts one by one and is reliable.

@dsiganos I implemented a generic count_accurate method for stores: https://github.com/Exxenoz/nano-node/commit/d58386ac33bc0fe632f2283e74c33b30c8763c4c

Will update your code later; If you don't mind I will include it here into this PR. Alternately, I could move the iterable_store implementation into another PR, if you think this PR is already too huge.